src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingMiner.java

```diff
@@ -169,18 +175,38 @@ public int getDepthLimit(final Level level)
     {
         if (this.getBuildingLevel() == 1)
         {
-            return MAX_DEPTH_LEVEL_1;
+            return normalizeMaxDepth(MAX_DEPTH_LEVEL_1, level);
         }
         else if (this.getBuildingLevel() == 2)
         {
-            return MAX_DEPTH_LEVEL_2;
+            return normalizeMaxDepth(MAX_DEPTH_LEVEL_2, level);
         }
-        else if (this.getBuildingLevel() >= 3)
+        else if (this.getBuildingLevel() == 3)
+        {
+            return normalizeMaxDepth(MAX_DEPTH_MAX, level);
+        }
+        else
         {
-            return level.getMinBuildHeight() + MAX_DEPTH_LEVEL_3;
+            return normalizeMaxDepth(-100, level);
         }
+    }
 
-        return MAX_DEPTH_LEVEL_0;
```

Comment:
Meanwhile this condition is no longer handled (building level 0 shouldn't dig at all) ... _hopefully_ that doesn't matter as the miner shouldn't start actually working until they hit level 1, but it might be safer to keep the original behaviour there.